system/handlers/admin/systemAlerts/invalidRuleEngineRules.cfc

```diff
@@ -0,0 +1,59 @@
+component {
+	property name="validationEngine"    inject="ValidationEngine";
+	property name="conditionService"    inject="RulesEngineConditionService";
+	property name="systemAlertsService" inject="SystemAlertsService";
+
+	private void function runCheck( required systemAlertCheck check ) {
+		var invalidRules  = [];
+		var ruleFilter    = "";
+		var ruleParams    = {};
+		var existingAlert = systemAlertsService.getAlert( type="invalidRuleEngineRules" );
+
+		if ( arrayLen( existingAlert.data.invalidRules ?: [] ) ) {
+			ruleFilter = "id in (:id) OR datecreated >= :datecreated";
```

Comment:
The date filter is only applied if there is an existing alert. So, if there is no existing alert, all rules will be checked every time this runs, which is not ideal! But all rules should be checked on startup, as the available rules may have changed...

This requires a small extra piece in the main system alerts code which I'll add in now: 
- A check should know how it was triggered (e.g. startup, manual, etc)
- A check should know when it eas last run, whether or not an alert was raised